synapse/module_api/__init__.py

```diff
@@ -73,14 +75,41 @@ def check_user_exists(self, user_id):
         """
         return self._auth_handler.check_user_exists(user_id)
 
-    def register(self, localpart):
-        """Registers a new user with given localpart
+    @defer.inlineCallbacks
+    def register(self, localpart, displayname=None, emails=[]):
+        """Registers a new user with given localpart and optional
+           displayname, email.
+
+        Args:
+            localpart (str): The localpart of the new user.
+            displayname (str|None): The displayname of the new user.
+            emails (List[str]): List of emails to assign to the new user.
 
         Returns:
             Deferred: a 2-tuple of (user_id, access_token)
         """
+        # Register the user
         reg = self.hs.get_registration_handler()
-        return reg.register(localpart=localpart)
+        user_id, access_token = yield reg.register(
+            localpart=localpart, default_display_name=displayname,
+        )
+
+        # Bind email address with the registered identity service
+        unix_secs = int(time.time())
```

Comment:
you want to use `hs.get_clock()`, mostly because it makes testing easier if anyone ever comes to unit test this